src/c++/perf_analyzer/profile_data_exporter.cc

```diff
@@ -151,6 +156,40 @@ ProfileDataExporter::AddResponseTimestamps(
   }
 }
 
+void
+ProfileDataExporter::AddRequestInputs(
+    rapidjson::Value& request_inputs_json,
+    const std::vector<RequestRecord::RequestInput>& request_inputs)
+{
+  for (const auto& request_input : request_inputs) {
+    for (const auto& input : request_input) {
+      const auto& name{input.first};
+      const auto& buf{input.second.data_.get()};
+      const auto& byte_size{input.second.size_};
+      const auto& data_type{input.second.data_type_};
+      rapidjson::Value name_json(name.c_str(), document_.GetAllocator());
+      rapidjson::Value input_json{};
+      if (buf != nullptr) {
+        if (data_type == "BYTES" || data_type == "JSON") {
```

Comment:
I don't like if/else switch statement chains that don't have a final else case. Are we sure these are the only datatypes that will ever be used? Please add a final `else { throw error }`

Reply:
I just added only the necessary ones for genai-perf but I can try to support all the types in [here](https://github.com/triton-inference-server/server/blob/main/docs/user_guide/model_configuration.md#datatypes).

For default behavior, I decided to set the json value of unsupported data types to `null` instead of throwing an error. The reason was that in that case, user at least gets to see the final json file but unsupported values as `null`, whereas if we throw an error, then user will wait for profile to finish, only to figure out that they cannot see the json file at all.

Reply:
Are you saying that is happening on line 171 (the constructor)? That input_json is null by default, but will still be added in line 187?

Reply:
Yes, empty rapidjson value is null by default.

Reply:
I still want a warning or something so we don't silently just drop information (although I would feel better if all types were supported)

Reply:
I agree. I've added warning message and a ticket to support other data types since I don't want this to block our visualization tasks.